lib/clean/convertToProperType.js

```diff
@@ -43,6 +43,19 @@ function convertToProperType(value, type) {
     if (typeof value === 'number') return new Date(value);
   }
 
+  // Convert to Boolean type
+  if (type === Boolean) {
+    if (typeof value === 'string') {
+      // Convert exact string 'true' and 'false' to true and false respectively
+      if (value === 'true') return true;
+      else if (value === 'false') return false;
+    } else {
+      if (typeof value === 'number') {
+        return Boolean(value);
+      }
+    }
```

Comment:
Combine the else and if into one and add a NaN check

```js
else if (typeof value === 'number' && !isNaN(value)) {
  return Boolean(value);
}
```

The NaN check is because Boolean(NaN) results in `false` but usually NaN is an error, so it's better to not do the type conversion, let it fail, and make sure the error is seen.

You can add a test for this passing in something like `Number('foo')` as the value, and expect it to be `NaN` (or might have to use `toEqual` because `NaN === NaN` is false).